### PR TITLE
Implement DeleteVirtualProductFile command

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Implement DeleteVirtualProductFile command & behat test. This command should delete the Entity and the actual file
| Type?             | refacto
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | part of #14772 
| How to test?      | CI :heavy_check_mark: 
| Possible impacts? | 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23363)

### CODE REVIEW

File: tests/Integration/Behaviour/Features/Scenario/Product/VirtualProductFile/delete_virtual_product_file.feature [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s product --tags virtual-product-file
```
Comment (Progi1984) [inline tests/Integration/Behaviour/Features/Scenario/Product/VirtualProductFile/delete_virtual_product_file.feature]:
```suggestion
# ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s product --tags delete-virtual-product-file
```

**Review by Progi1984 — COMMENTED** [review]
A small feedback

**Review by jolelievre — COMMENTED** [review]
Everything is ok for me except one question, shouldn't the file be removed from hard drive as well?

**Comment by zuk3975** [comment]
> Everything is ok for me except one question, shouldn't the file be removed from hard drive as well?

thanks you've spotted a problem. I was thinking that Object model handles it, but apparently not (until we pas $deleteFile = true).
So Im adding WIP again and gonna update the PR by introducing some file remover service in Adapter/File and maybe adjusting behats to spot filesystem changes :muscle:

**Review by jolelievre — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thank you @zuk3975

**Pull request merged**